### PR TITLE
Add Largest Sum Contiguous Subarray in TypeScript

### LINKED ISSUE #727: Largest Sum Contiguous Sub-Array in Typescript
This is an open issue for the Largest Sum Contiguous Sub-Array algorithm implementation in Typescript. If you wish to contribute, just make some noise and we will assign you.

### PR DESCRIPTION
**Pull request by bmpj13**
**Fixes #727**

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [x] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [x] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [x] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [x] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [x] This pull request shall only be merged once the build passes.

-   [x] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: algorithms/ar-lscsa/typescript/largest_sum_contiguous_subarray.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        if (localMax < 0) localMax = 0;
+    }
+
+    return globalMax;
+}
+
+function main () {
+	let input = [2, 3, 0, 4];
+    const largestSum = largestSumContiguousSubarray(input);
+    console.log(largestSum);
```
Comment (RavinduSachintha) [inline algorithms/ar-lscsa/typescript/largest_sum_contiguous_subarray.ts:18]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6391064)

File: algorithms/ar-lscsa/typescript/largest_sum_contiguous_subarray.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+function largestSumContiguousSubarray (arr: number[]) {
+	let globalMax = Number.MIN_VALUE;
+    let localMax = 0;
+    
+    for (const num of arr) {
+        localMax += num;
+
+        if (globalMax < localMax) globalMax = localMax;
+        if (localMax < 0) localMax = 0;
```
Comment (RavinduSachintha) [inline algorithms/ar-lscsa/typescript/largest_sum_contiguous_subarray.ts:9]: Codacy found an issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6391064)

File: algorithms/ar-lscsa/typescript/largest_sum_contiguous_subarray.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+function largestSumContiguousSubarray (arr: number[]) {
+	let globalMax = Number.MIN_VALUE;
+    let localMax = 0;
+    
+    for (const num of arr) {
+        localMax += num;
+
+        if (globalMax < localMax) globalMax = localMax;
```
Comment (RavinduSachintha) [inline algorithms/ar-lscsa/typescript/largest_sum_contiguous_subarray.ts:8]: Codacy found an issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6391064)

File: algorithms/ar-lscsa/typescript/largest_sum_contiguous_subarray.ts (hunk 1 above)
Comment (RavinduSachintha) [inline algorithms/ar-lscsa/typescript/largest_sum_contiguous_subarray.ts:18]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6391064)

File: algorithms/ar-lscsa/typescript/largest_sum_contiguous_subarray.ts (hunk 2 above)
Comment (RavinduSachintha) [inline algorithms/ar-lscsa/typescript/largest_sum_contiguous_subarray.ts:9]: Codacy found an issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6391064)

File: algorithms/ar-lscsa/typescript/largest_sum_contiguous_subarray.ts (hunk 3 above)
Comment (RavinduSachintha) [inline algorithms/ar-lscsa/typescript/largest_sum_contiguous_subarray.ts:8]: Codacy found an issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6391064)

File: algorithms/ar-lscsa/typescript/largest_sum_contiguous_subarray.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    let globalMax = Number.MIN_VALUE;
+    let localMax = 0;
+
+    for (const num of arr) {
+        localMax += num;
+
+        if (globalMax < localMax) {
+            globalMax = localMax;
+        }
+        if (localMax < 0) {
```
Comment (RavinduSachintha) [inline algorithms/ar-lscsa/typescript/largest_sum_contiguous_subarray.ts:11]: Codacy found an issue: [Irregular whitespace not allowed.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6391064)

**Review by Danushka96 — APPROVED** [review]

**Comment by Danushka96** [comment]
@all-contributors please add @bmpj13  for code and doc

**Comment by Danushka96** [comment]
Hi @bmpj13 
Thanks for your contribution, keep it up!

**Pull request merged**